### PR TITLE
[TwigBridge] lint all templates from configured Twig paths if no argument was provided

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/issues/33443
| License       | MIT
| Doc PR        | -

All details in PR title and related issue https://github.com/symfony/symfony/issues/33443.

### CODE REVIEW

File: src/Symfony/Bridge/Twig/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ CHANGELOG
  * marked all classes extending twig as `@final`
  * deprecated to pass `$rootDir` and `$fileLinkFormatter` as 5th and 6th argument respectively to the 
    `DebugCommand::__construct()` method, swap the variables position.
+ * the `LintCommand` lint all the templates stored in all configured Twig paths if none argument is provided
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/CHANGELOG.md]: lints all the templates?

**Review by fabpot — APPROVED** [review]

File: src/Symfony/Bridge/Twig/CHANGELOG.md (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Bridge/Twig/CHANGELOG.md]: fixed

**Comment by fabpot** [comment]
Thank you @yceruto.

**Comment by nicolas-grekas** [comment]
There's a conflict here: when no arguments was provided, the command also reads from STDIN.
So now, it reads from STDIN, and if there is nothing there, reads from the default template.
This has been caught in https://github.com/php/php-src/pull/4672

This creates an ambiguous situation - maybe one did pipe nothing but doesn't expect the default template dir to be linted.

I'd suggest resolving the ambiguity by reading from STDIN only when explicitly asked for. Passing `-` as argument could the way. And we could trigger a deprecation for now.

**Comment by nicolas-grekas** [comment]
I think we should revert this and trigger the deprecation instead...
Then in 5.0 we would introduce this behavior.

**Comment by yceruto** [comment]
Maybe resolved by https://github.com/php/php-src/pull/4684 ? 

Not sure also if reverting is the solution... because before these changes this statement `0 !== ftell(STDIN)` is always called if no dir or file was provided, the difference from now on is that you have a test case. If there is something to fix, I guess it should be target to 3.4 branch?

**Comment by nicolas-grekas** [comment]
Ok cool. Thanks for having a look.

**Comment by nikic** [comment]
How is this code actually supposed to work? Under what circumstances do you want to use the stdin input and when not? Should it be used only with `<` or also with `|`? The aforementioned php-src commit will make it so the stdin codepath is only used for the `<` case, but not for `|` or tty. Previously (I think) stdin would have been read for `|` as well, and ttys on Windows.

**Comment by nikic** [comment]
Test is still hanging despite the change: https://dev.azure.com/phpazuredevops/PHP/_build/results?buildId=2477

**Comment by nikic** [comment]
Unfortunately I'm not able to reproduce this issue locally. I've tried various variations of what to attach to stdin, but nothing produces the hang we see on azure.

Possibly this just needs a `@group tty` on the test.

**Comment by nicolas-grekas** [comment]
tty group added in #33490

**Pull request merged**